### DISCUSSION TITLE
self-hosted problems with sentry

### DISCUSSION THREAD

**Discussion by jsarola** [plausible/analytics#3166] [Self-Hosted Support]
Hi,

I've installed the last version (v2.0) and sentry is configured correctly with SENTRY_DSN, and the problem is when plausible try to send and event:
```text
Failed to send Sentry event. Unable to encode JSON Sentry error .....

description: "Jason.Encoder protocol must always be explicitly implemented.\n\nIf you own the struct, you can derive the implementation specifying which fields should be encoded to JSON:\n\n    @derive {Jason.Encoder, only: [....]}\n    defstruct ...\n\nIt is also possible to encode all fields, although this should be used carefully to avoid accidentally leaking private information when new fields are added:\n\n    @derive Jason.Encoder\n    defstruct ...\n\nFinally, if you don't own the struct you want to encode to JSON, you may use Protocol.derive/3 placed outside of any module:\n\n    Protocol.derive(Jason.Encoder, NameOfTheStruct, only: [...])\n    Protocol.derive(Jason.Encoder, NameOfTheStruct)\n"}
```

Is it a configuration issue? Someone have found the same error?

Regards,

**Comment by ruslandoga** [comment]
👋 @jsarola 

Can you please post the event / exception that Sentry is trying to send? If it's an exception, it should show up just above this error in the logs.

**Reply by jsarola** [reply]
Hi,
Sorry for the delay, this is the last message that appears in console, with the same error:
`
16:41:40.027 request_id=F3Og_no1gcB0qXgAAxEx [warning] Failed to send Sentry event. Unable to encode JSON Sentry error - %Protocol.UndefinedError{protocol: Jason.Encoder, value: %Plausible.Ingestion.Request{id: nil, remote_ip: "172.20.0.1", user_agent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.5.2 Mobile/15E148 Safari/604.1", event_name: "pageview", uri: %URI{scheme: "https", authority: "www.domain.local", userinfo: nil, host: "www.domain.local", port: 443, path: "/es/", query: nil, fragment: nil}, hostname: "www.domain.local", referrer: "https://another.domain.local/", domains: ["www.domain.local"], hash_mode: nil, pathname: "/es/", props: %{}, query_params: nil, timestamp: ~N[2023-07-20 16:41:40]}, description: "Jason.Encoder protocol must always be explicitly implemented.\n\nIf you own the struct, you can derive the implementation specifying which fields should be encoded to JSON:\n\n    @derive {Jason.Encoder, only: [....]}\n    defstruct ...\n\nIt is also possible to encode all fields, although this should be used carefully to avoid accidentally leaking private information when new fields are added:\n\n    @derive Jason.Encoder\n    defstruct ...\n\nFinally, if you don't own the struct you want to encode to JSON, you may use Protocol.derive/3 placed outside of any module:\n\n    Protocol.derive(Jason.Encoder, NameOfTheStruct, only: [...])\n    Protocol.derive(Jason.Encoder, NameOfTheStruct)\n"}
`

Regards,

**Reply by ruslandoga** [reply]
@jsarola thank you! Seems like the error is caused by adding `Request` struct to Sentry context here https://github.com/plausible/analytics/blob/16846b16c8e513f72d7c23459ee879d9249f556d/lib/plausible_web/controllers/api/external_controller.ex#L15

@aerosol @vinibrsl should we `inspect(request)` or implement a Jason encoder for `%Request{}`? Otherwise no errors can be reported for this controller action.

**Reply by ruslandoga** [reply]
Quickfix with `inspect` https://github.com/plausible/analytics/pull/3180

**Reply by aerosol** [reply]
JSON encoder would be the best, otherwise there's not much in sentry it seems.
